lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -302,16 +312,41 @@ def write(chunk)
             entry['metadata']['severity'] = 'DEFAULT'
           end
 
+          if @service_name == CONTAINER_SERVICE
+            # TODO(a-robinson): Consider falling back to parsing the namespace,
+            # pod, and container names out of the log name if we don't have this
+            # information, which depends on the kubernetes_metadata_filter
+            # filter plugin.
+            labels = {}
+            field_to_label(record['kubernetes'], 'namespace_id', labels,
+                                  "#{CONTAINER_SERVICE}/namespace_id")
+            field_to_label(record['kubernetes'], 'namespace_name', labels,
+                                  "#{CONTAINER_SERVICE}/namespace_name")
+            field_to_label(record['kubernetes'], 'pod_id', labels,
```

Comment:
Why is pod id & namespace id required? I assume these are the respective uids? Is name but sufficient? Are names mutable now?